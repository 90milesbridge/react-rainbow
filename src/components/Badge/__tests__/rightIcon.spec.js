import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import RightIcon from './../rightIcon';

describe('<RightIcon in the Badge component/>', () => {
    it('should not have children when the iconName is not passed', () => {
        const component = mount(
            <RightIcon position="right" />,
        );
        expect(component.children().length).toBe(0);
    });
    it('should not have children when the position is other than right', () => {
        const component = mount(
            <RightIcon iconName="utility:world" position="left" />,
        );
        expect(component.children().length).toBe(0);
    });
    it('should have the right class names when the label is not passed', () => {
        const component = renderer.create(
           <RightIcon iconName="utility:world" position="right" />,
       );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when the label is passed', () => {
        const component = renderer.create(
           <RightIcon label="testing RightIcon" iconName="utility:world" position="right" />,
       );
        expect(component).toMatchSnapshot();
    });
});
