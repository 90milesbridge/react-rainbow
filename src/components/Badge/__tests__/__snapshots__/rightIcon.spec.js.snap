// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightIcon in the Badge component/> should have the right class names when the label is not passed 1`] = `
<span
  className="slds-icon_container"
  style={Object {}}
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_xx-small slds-icon-text-default"
    style={Object {}}
  >
    <use
      xlinkHref="/icons/utility-sprite/svg/symbols.svg#world"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    />
  </svg>
</span>
`;

exports[`<RightIcon in the Badge component/> should have the right class names when the label is passed 1`] = `
<span
  className="slds-icon_container slds-m-left_xx-small"
  style={Object {}}
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_xx-small slds-icon-text-default"
    style={Object {}}
  >
    <use
      xlinkHref="/icons/utility-sprite/svg/symbols.svg#world"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    />
  </svg>
</span>
`;
